08-transport.md

```diff
@@ -253,7 +256,9 @@ and 16 bytes for the `poly1305` tag.
     * The raw bytes of the remote party's ephemeral public key (`re`) are to be
       deserialized into a point on the curve using affine coordinates as encoded
       by the key's serialized composed format.
-3. If `v` is an unrecognized handshake version, then the responder MUST
+3. If `v` is an unrecognized handshake version, and the responder supports
```

Comment:
So the version here was intended to be used to swap out for new _crypto_ in the future. 

When I saw the title of this PR, I thought this was instead adding a new address type (ws) to the node announcement message's set of addresses, and leaving everything else unmodified. This would let nodes advertise they support ws (maybe on a different interface/port) and keep everything else in place.

Reply:
IMO the address route also means that the message encryption layer doesn't need to be aware of what's being used to transport it. Instead nodes may optionally (if the include the new addr type) start to listen on a new port, expecting a WS handshake as the first step.

Reply:
Yeah, this is a bit awkward since "this port supports ws" is orthogonal to the address format, so I just made it a 2-byte port number, implying it's an alternate to the other address types which have a port.

And yeah, that's exactly what happens: you get a HTTP incoming, upgrades to ws, then you transport lightning over the top (we ignore the ws framing and treat it as a stream, but that's trivial).

Reply:
> Yeah, this is a bit awkward since "this port supports ws" is orthogonal to the address format, so I just made it a 2-byte port number, implying it's an alternate to the other address types which have a port.

Yeah so what I'm suggesting is to effectively just treat it as a new address type, since that lets nodes handle it fully indepndently (they can bridge or make it native). 

The difference here is (also from looking at your current impl), is that the noise layer doesn't need to "peak" into the handshake state to understand it. Instead we get a clean separation, just like it doesn't need to know anything about Tor when that's used, as in the context of the language runtime, it likely just gets some "connection-like" object to use once that transport handshake has been completed.